Update Core Node Admin Guide (#376)
stellar/stellar-docs: network/core-node/admin-guide/publishing-history-archives.mdx

sidebar_position: 50
---

> If you want to run a [Full Validator](../README.mdx#full-validator) or an [Archiver](../README.mdx#archiver), you need to set up your node to publish a history archive. You can host an archive using a blob store such as Amazon's S3 or Digital Ocean's spaces, or you can simply serve a local archive directly via an HTTP server such as Nginx or Apache. If you're setting up a [Basic Validator](../README.mdx#basic-validator), you can skip this section. No matter what kind of node you're planning to run, make sure to set it up to `get` history, which is covered in [Environment Preparation](./environment-preparation.mdx).

Reviewer: I'd remove "Archiver."

## Caching and History Archives

> You can significantly reduce the data transfer costs associated with running a public History archive by using common caching techniques or a CDN. Three simple rules apply to caching the History archives:

Reviewer: Do we want to point out that bandwidth costs for egress are usually the most expensive part of running a validator?  And that caching is crucial to keep those costs down?  I think something along those lines might be helpful

}
```

## Amazon S3 History Archive

> Now, let's demonstrate aconfiguration where your node stores its history files using Amazon's S3 service. can then publish that history using an Amazon S3 static site, or we can again use nginx for our webserver software. This time, using nginx, we'll include some proxy and CDN configuration, as well.

Reviewer: ```suggestion
Now, let's demonstrate a configuration where your node stores its history files using Amazon's S3 service.  You can then publish that history using an Amazon S3 static site, or again use nginx for your webserver software. This time, using nginx, we'll include some proxy and CDN configuration, as well.
```

}
```

## Backfilling a History Archive

Given the choice, it's best to configure your history archive prior to your node's initial sync with an existing network. That way your validator's history publishes as you join and subsequently sync with the network.

> However, if you have _not_ published an archive during the node's initial sync, it's still possible to use the [stellar-archivist](https://github.com/stellar/go/tree/master/tools/stellar-archivist) command line tool to mirror, scan, and repair existing archives. Using the [SDF package repositories](https://github.com/stellar/packages), you can install `stellar-archivist` by running:

Reviewer: These paragraphs about using stellar-archivist seem to belong to the ##Complete History Archive section below.  Possible to consolidate?

Author: I've shuffled this around a bit, but I think you're right: it could definitely benefit from some re-thinking/consolidating. I'll let it marinade overnight, and see what I can make happen tomorrow.

mkdir="mkdir -p /mnt/xvdf/stellar-core-archive/node_001/{0}"
```

Next, you must run Stellar Core's `new-hist` command to create and initialize the local archive. (This is done as the `stellar` user.)

```bash
sudo -u stellar stellar-core --conf /etc/stellar/stellar-core.cfg new-hist local
```

Now you can start your Stellar Core instance again:

```bash
systemctl start stellar-core # modify this if not using systemctl
```

As you allow your node to join the network again, you can watch it start publishing a few checkpoints to the newly created archive.

```log
2019-04-25T12:30:43.275 GDUQJ [History INFO] Publishing 1 queued checkpoints [16895-16895]: Awaiting 0/0 prerequisites of: publish-000041ff
```

> At this stage your validator is successfully publishing its history, which enables other users to join the network using your archive (although it won't allow them to `CATCHUP_COMPLETE=true` using your node, as the archive only has partial network history).

Reviewer: ```suggestion
At this stage your validator is successfully publishing its history, which enables other users to join the network using your archive.
```

Reviewer: I think the CATCHUP_COMPLETE line may be misleading (since that's a setting that replays history)
